api/RestfulService.php

```diff
@@ -164,37 +170,18 @@ public function request($subURL = '', $method = "GET", $data = null, $headers =
 			$curlOptions + (array)$this->config()->default_curl_options,
 			$this->getBasicAuthString()
 		));
-
-		// Check for unexpired cached feed (unless flush is set)
-		//assume any cache_expire that is 0 or less means that we dont want to
-		// cache
-		if($this->cache_expire > 0 && !isset($_GET['flush'])
-				&& @file_exists($cache_path)
-				&& @filemtime($cache_path) + $this->cache_expire > time()) {
-
-			$store = file_get_contents($cache_path);
-			$response = unserialize($store);
-
-		} else {
+		$cache = SS_Cache::factory(__CLASS__, 'Output', array(
```

Comment:
I have hardcoded "RestfulService" even if I had mixed feelings about doing that.
NB: my last squashed commit also fixes a bug introduced by the refactoring for the Flushable feature: a check was done against a true value for self::$flush before using the cached response, while we should be able to use the cached value without having flush=1 in the url